src/app/loadout-builder/generated-sets/GeneratedSets.tsx

```diff
@@ -193,18 +194,24 @@ export default class GeneratedSets extends React.Component<Props, State> {
   };
 
   private addLockedItemType = (item: LockedItemType) => {
-    const { lockedMap, onLockedMapChanged } = this.props;
-    onLockedMapChanged({
-      ...lockedMap,
-      [item.bucket.hash]: addLockedItem(item, lockedMap[item.bucket.hash]),
+    const { lockedMap, lbDispatch } = this.props;
+    lbDispatch({
+      type: 'lockedMapChanged',
+      lockedMap: {
+        ...lockedMap,
+        [item.bucket.hash]: addLockedItem(item, lockedMap[item.bucket.hash]),
```

Comment:
It'd be neat to dispatch an `addLockedItem` action instead of recomputing the whole locked map here.

Reply:
Can do, I plan to move all the complex state calculations once the worker is done.